src/clp_ffi_py/ir/native/PyKeyValuePairLogEvent.hpp

```diff
@@ -0,0 +1,88 @@
+#ifndef CLP_FFI_PY_IR_NATIVE_PYKEYVALUEPAIRLOGEVENT_HPP
+#define CLP_FFI_PY_IR_NATIVE_PYKEYVALUEPAIRLOGEVENT_HPP
+
+#include <clp_ffi_py/Python.hpp>  // Must always be included before any other header files
+
+#include <gsl/gsl>
+
+#include <clp/ffi/KeyValuePairLogEvent.hpp>
+
+#include <clp_ffi_py/PyObjectUtils.hpp>
+
+namespace clp_ffi_py::ir::native {
+/**
+ * A PyObject structure functioning as a Python-compatible interface to retrieve a key-value pair
+ * log event. The underlying data is pointed by `m_kv_pair_log_event`.
+ */
+class PyKeyValuePairLogEvent {
+public:
+    // Delete default constructor to disable direct instantiation.
+    PyKeyValuePairLogEvent() = delete;
+
+    // Delete copy/move constructors and assignments
+    PyKeyValuePairLogEvent(PyKeyValuePairLogEvent const&) = delete;
+    PyKeyValuePairLogEvent(PyKeyValuePairLogEvent&&) = delete;
+    auto operator=(PyKeyValuePairLogEvent const&) -> PyKeyValuePairLogEvent& = delete;
+    auto operator=(PyKeyValuePairLogEvent&&) -> PyKeyValuePairLogEvent& = delete;
+
+    // Destructor
+    ~PyKeyValuePairLogEvent() = default;
+
+    /**
+     * Initializes the underlying data with the given inputs. Since the memory allocation of
+     * `PyKeyValuePairLogEvent` is handled by CPython's allocator, cpp constructors will not be
+     * explicitly called. This function serves as the default constructor initialize the underlying
+     * key-value pair log event. It has to be manually called whenever creating a new
+     * `PyKeyValuePairLogEvent` object through CPython APIs.
+     * @param kv_pair_log_event
+     * @return true on success.
+     * @return false on failure with the relevant Python exception and error set.
+     */
+    [[nodiscard]] auto init(clp::ffi::KeyValuePairLogEvent kv_pair_log_event) -> bool;
+
+    /**
+     * Initializes the pointers to nullptr by default. Should be called once the object is
+     * allocated.
+     */
+    auto default_init() -> void { m_kv_pair_log_event = nullptr; }
+
+    /**
+     * Releases the memory allocated for underlying data fields.
+     */
+    auto clean() -> void {
```

Comment:
This is following the standard we made for Python object classes. `init`, `default_init`, `clean`, and some other static members are the same for all Python objects. I think `deinit` might be a better name though, but shall we only apply these to newly added classes? @kirkrodrigues what do u think?

Reply:
Since the existing code is consistent, I would prefer to maintain consistency and then address the rename in a single PR.

Btw, didn't we consider deinit before?

Reply:
sure, doesn't need to be in this PR.

Reply:
https://github.com/y-scope/clp-ffi-py/pull/3#discussion_r1267397537